examples/core-hello-world/src/test/scala/com/mesosphere/usi/examples/CoreHelloWorldFrameworkTest.scala

```diff
@@ -31,6 +47,68 @@ class CoreHelloWorldFrameworkTest extends AkkaUnitTest with MesosClusterTest wit
     }
   }
 
+  "CoreHelloWorldFramework should adhere to at most once launch guarantee during a crash-recovery" in {
+    Given("a scheduler graph that uses a persistence layer")
+    val conf = loadConfig
+    val frameworkInfo = CoreHelloWorldFramework.buildFrameworkInfo
+    val customPersistenceStore = InMemoryPodRecordRepository()
+    customPersistenceStore.readAll().futureValue.size shouldBe 0
+    val (mesosClient, scheduler) = CoreHelloWorldFramework.buildGraph(conf, customPersistenceStore, frameworkInfo)
+
+    And("an initial pod spec is launched")
+    val specsSnapshot = CoreHelloWorldFramework.generateSpecSnapshot
+    val (_, sub1) = runGraphAndFeedSnapshot(scheduler, specsSnapshot)
+
+    Then("receive an empty snapshot followed by other state events")
+    val e1 = sub1.requestNext()
+    e1 shouldBe a[StateSnapshot]
+    e1.asInstanceOf[StateSnapshot].podRecords shouldBe empty
+    val e2 = sub1.requestNext()
```

Comment:
I'm not a big fan of `eN` kind of names. Wdyt about making them more descriptive e.g. `snapshot`, `record`, `status` etc. or maybe reusing `var event`?

Reply:
Yeah, actually, lets do inside here:

```
inside(sub1.requestNext) {
  case snapshot: Snapshot =>
    snapshot.podRecords shouldBe empty
}
```